sql/moz-fx-data-shared-prod/telemetry_derived/core_clients_first_seen_v1/query.sql

```diff
@@ -1,20 +1,27 @@
-WITH clients_today AS (
+WITH _current AS (
   SELECT DISTINCT
     client_id
   FROM
     telemetry.core
   WHERE
     DATE(submission_timestamp) = @submission_date
+),
+_previous AS (
+  SELECT
+    *
+  FROM
+    telemetry_derived.core_clients_first_seen_v1
+  WHERE
+    first_seen_date > "2010-01-01"
 )
 SELECT
   client_id,
   @submission_date AS first_seen_date
 FROM
-  clients_today
+  _current
 LEFT JOIN
-  telemetry_derived.core_clients_first_seen_v1 AS cfs
+  _previous
 USING
   (client_id)
 WHERE
-  cfs.first_seen_date < @submission_date
```

Comment:
This is the problematic statement; it's required because of the partition filter, but also happens to filter out null clients in the join (e.g. new clients to the table). I add this filter in a cte so it's not confounded here.

Reply:
Makes sense. Thanks for diving in and figuring this out.